llvm/lib/Target/AArch64/AArch64ISelLowering.h

```diff
@@ -135,6 +135,8 @@ enum NodeType : unsigned {
   UDIV_PRED,
   UMAX_PRED,
   UMIN_PRED,
+  FAMAX_PRED,
+  FAMIN_PRED,
```

Comment:
Is it necessary to create dedicated AArch64ISD nodes? We normally reserve that for cases where additional DAG combines are available or for when the operation is hard to match during ISel, neither of which apply in this case?

Reply:
`fmin`/`fmax` have representations in LLVM IR and equivalent ISD nodes, which is why the `_PRED` variants exist.  This is not true for `famin`/`famax`, which are just two existing operations joined together.

Reply:
How about:
```
  case Intrinsic::aarch64_sve_fmin_u:
    return DAG.getNode(AArch64ISD::FMIN_PRED, SDLoc(N), N->getValueType(0),
                       N->getOperand(1), N->getOperand(2), N->getOperand(3));
```

Should it also not use a custom ISD node? Or we use here it because we have it anyway? (for lowering `ISD::FMINIMUM` ?)

Reply:
Yep, the latter. We use it here because the ISD node exists for other reasons and so was convenient to reuse here.

In general the main reason to create `FAMAX_PRED` is when there's optimisation to be had.  In this instance I think it's more likely for us break the operation into separate FABS_PRED/FMIN_PRED nodes to let the combiner at it, given it'll be easy to reconstruct the `famin` during isel.